app/models/concerns/test_track/identity.rb

```diff
@@ -46,6 +46,17 @@ def test_track_identifier(identifier_type, identifier_value_method)
             end
           end
         end
+
+        define_method :test_track_sign_up! do
+          discriminator = TestTrack::IdentitySessionDiscriminator.new(self)
+          identifier_value = send(identifier_value_method)
+
+          if discriminator.participate_in_unauthenticated_session?
+            discriminator.controller.send(:test_track_session).sign_up! identifier_type, identifier_value
+          else
+            TestTrack::OfflineSession.create_visitor_for(identifier_type, identifier_value)
```

Comment:
Here's the code on the server side that a subsequent attempt to `with_visitor_for` would call: https://github.com/Betterment/test_track/blob/master/app/models/visitor_lookup.rb#L20-L27 which leads me to believe nooping here would be totally safe.